WeightProducer/python/MCSampleValues.py

```diff
@@ -63,493 +63,1039 @@ class MCSampleValuesHelper():
-            "CrossSection" : XSValues(XS_13TeV=125.9),
+            "CrossSection" : XSValues(
+                XS_2016=131.8, XSSource_2016="XSDB (LO)",
+                XS_2017=124.6, XSSource_2017="GenXSecAnalyzer",
+                XS_2018=124.7, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "GJets_DR-0p4_HT-600ToInf" : {
-            "CrossSection" : XSValues(XS_13TeV=43.36),
+            "CrossSection" : XSValues(
+                XS_2016=44.27, XSSource_2016="XSDB (LO)",
+                XS_2017=40.65, XSSource_2017="GenXSecAnalyzer",
+                XS_2018=40.46, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "ZJetsToNuNu_Zpt-100to200" : {
-            "CrossSection" : XSValues(XS_13TeV=43.5543),
+            "CrossSection" : XSValues(
+                XS_2016=35.99, XSSource_2016="XSDB (LO)",
+            ),
         },
         "ZJetsToNuNu_Zpt-200toInf" : {
-            "CrossSection" : XSValues(XS_13TeV=5.28654),
-        },
-        "ST_s-channel_4f_hadronicDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=6.96),
+            "CrossSection" : XSValues(
+                XS_2016=4.201, XSSource_2016="XSDB (LO)",
+            ),
         },
         # single top: NoFullyHadronicDecays xsec scaled by BF for non-fully-hadronic (1-(1-3*0.108)^2)
         "ST_s-channel_4f_leptonDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=3.34),
+            "CrossSection" : XSValues(
+                XS_13TeV=3.36, XSSource_13TeV="https://twiki.cern.ch/twiki/bin/viewauth/CMS/SummaryTable1G25n (NLO)",
+                XS_2016=3.365, XSSource_2016="XSDB (unknown)",
+                XS_2017=3.74,  XSSource_2017="XSDB (unknown)",
+                XS_2018=3.740, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "ST_s-channel_4f_InclusiveDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=10.32),
+            "CrossSection" : XSValues(
+                XS_13TeV=10.12, XSSource_13TeV="XSDB (unknown)",
+            ),
         },
         "ST_t-channel_top_4f_inclusiveDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=136.02),
+            "CrossSection" : XSValues(
+                XS_13TeV=136.02, XSSource_13TeV="Unknown",
+                XS_2017=113.3,   XSSource_2017="XSDB (NLO)",
+            ),
         },
         "ST_t-channel_top_4f_InclusiveDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=136.02),
+            "CrossSection" : XSValues(
+                XS_13TeV=136.02, XSSource_13TeV="Unknown",
+                XS_2017=113.3,   XSSource_2017="XSDB (NLO)",
+            ),
         },
         "ST_t-channel_antitop_4f_inclusiveDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=80.95),
+            "CrossSection" : XSValues(
+                XS_13TeV=80.95, XSSource_13TeV="Unknown",
+                XS_2017=67.91,  XSSource_2017="XSDB (NLO)",
+            ),
         },
         "ST_t-channel_antitop_4f_InclusiveDecays" : {
```

Comment:
we should figure out a better way to avoid duplicating all the xsec values just because of bad GEN naming conventions

Reply:
Suggestions?

Reply:
add an "alternate name" field?

Reply:
or perhaps even easier to implement:
just define the alternate names as copies of the existing entries at the bottom of the file, after the dictionary is defined

Reply:
Implemented.